### PR TITLE
poc for finding cves

### PR DESCRIPTION
**Pull request by maleck13**
This change will find exactly what tag it running and list any CVEs effecting that tag

### CODE REVIEW

**Comment by maleck13** [comment]
@jasonmadigan  @alanmoran Found how to get the CVEs for an image running in the cluster. Still very much PoC

File: cmd/cli/main.go [hunk 1]
```diff
@@ -12,8 +13,11 @@ import (
 	"os"
 	"regexp"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"strings"
 )
 
+//curl 'https://rhcc-api.redhat.com/rest/v1/repository/registry.access.redhat.com/amq7%252Famq-online-1-api-server/images' -H 'accept: application/json' -H 'Origin: https://access.redhat.com' -H 'Sec-Fetch-Mode: cors' --compressed
```
Comment (jasonmadigan) [inline cmd/cli/main.go:19]: @maleck13 re this API - any idea how "supported" this is? is this the one from the RH security api, or an API powering the portal? Mostly just curious if it'll a) break often or b) go away at some point and inconvenience us

**Review by jasonmadigan — COMMENTED** [review]
very neat, @maleck13 

could you attach/show general output? have not tested/run it locally yet

File: cmd/cli/main.go (hunk 1 above)
Comment (maleck13) [reply cmd/cli/main.go:19]: I don't know who owns this api. Would like to find out though.

**Comment by maleck13** [comment]
The output is pretty gross at the moment. 

```
./cli check openshift-enmasse api-server
regex api-server
2019/09/30 11:14:08 not checking  address-space-controller does not match regex
2019/09/30 11:14:08 not checking  admin.zig8wi9ekw does not match regex
2019/09/30 11:14:08 deployment container name  api-server pod container name  api-server
parsing image  registry.redhat.io/amq7/amq-online-1-api-server:1.2
reg parts [registry.redhat.io amq7 amq-online-1-api-server:1.2]
image details amq-online-1-api-server 1.2 sha @sha256:2d593d562306ae62523c15029ba589f4de71b3370cbf1f985ed4efef39017631
2019/09/30 11:14:08 not checking  console does not match regex
2019/09/30 11:14:08 not checking  enmasse-operator does not match regex
2019/09/30 11:14:08 not checking  keycloak does not match regex
2019/09/30 11:14:08 not checking  service-broker does not match regex
---- CHECK RESULT ----
Checking image registry.redhat.io/amq7/amq-online-1-api-server:1.2
registry  registry.redhat.io org  amq7/amq-online-1-api-server  image  amq-online-1-api-server
Running hash:  sha256:1839b031b2a3bba8376aabc3ce54b12291e730e8cb2bc41e700d97a9c471cbc1
Latest Registry  hash:  sha256:343f921fe5e328a526513bc8ab6a4f25e962a4f5441d8d445b38b49e9412ef72 err <nil>
!XX RUNNING IMAGE IS NOT!! UPTO DATE XX!
-- END RESULT ---
calling images url  https://rhcc-api.redhat.com/rest/v1/repository/registry.access.redhat.com/amq7%252Famq-online-1-api-server/images
found tags in registry  [latest 1.2 1.2-7 1.2-6 1.1 1.1-16 1.1-14 1.0-7.1554788698 1.0 1.0-8 1.0-7.1553789970 1.2-5 1.0-6 1.1-5 1.1-9 1.1-11 1.0-7 1.0-3]
actual image tag running in cluster is  1.2-6
checking CVEs
calling image url  https://rhcc-api.redhat.com/rest/v1/repository/registry.access.redhat.com/amq7%252Famq-online-1-api-server/images/1.2-6?architecture=
found cves  [{Low [] 2019:1880 CVE-2018-14618} {Moderate [] 2019:1884 CVE-2019-3862} {Moderate [] 2019:2030 CVE-2019-9947} {Moderate [] 2019:2030 CVE-2019-9948} {Moderate [] 2019:2030 CVE-2018-14647} {Moderate [] 2019:2030 CVE-2019-9740} {Moderate [] 2019:2030 CVE-2019-5010} {Moderate [] 2019:2052 CVE-2016-3616} {Moderate [] 2019:2052 CVE-2018-11213} {Moderate [] 2019:2052 CVE-2018-14498} {Moderate [] 2019:2052 CVE-2018-11212} {Moderate [] 2019:2052 CVE-2018-11813} {Moderate [] 2019:2052 CVE-2018-11214} {Moderate [] 2019:2075 CVE-2018-12641} {Moderate [] 2019:2075 CVE-2018-12697} {Moderate [] 2019:2075 CVE-2018-1000876} {Moderate [] 2019:2079 CVE-2018-15854} {Moderate [] 2019:2079 CVE-2018-15857} {Moderate [] 2019:2079 CVE-2018-15859} {Moderate [] 2019:2079 CVE-2018-15853} {Moderate [] 2019:2079 CVE-2018-14599} {Moderate [] 2019:2079 CVE-2018-15864} {Moderate [] 2019:2079 CVE-2018-15863} {Moderate [] 2019:2079 CVE-2018-14600} {Moderate [] 2019:2079 CVE-2018-15856} {Moderate [] 2019:2079 CVE-2018-15855} {Moderate [] 2019:2079 CVE-2018-14598} {Moderate [] 2019:2079 CVE-2018-15862} {Moderate [] 2019:2079 CVE-2018-15861} {Moderate [] 2019:2091 CVE-2018-16888} {Moderate [] 2019:2091 CVE-2018-16866} {Moderate [] 2019:2091 CVE-2018-15686} {Moderate [] 2019:2118 CVE-2016-10739} {Moderate [] 2019:2136 CVE-2019-3858} {Moderate [] 2019:2136 CVE-2019-3861} {Low [] 2019:2159 CVE-2018-18384} {Low [] 2019:2181 CVE-2018-16842} {Moderate [] 2019:2189 CVE-2018-1122} {Low [] 2019:2197 CVE-2019-7150} {Low [] 2019:2197 CVE-2018-18521} {Low [] 2019:2197 CVE-2018-18520} {Low [] 2019:2197 CVE-2019-7149} {Low [] 2019:2197 CVE-2018-16062} {Low [] 2019:2197 CVE-2018-16403} {Low [] 2019:2197 CVE-2018-16402} {Low [] 2019:2197 CVE-2019-7664} {Low [] 2019:2197 CVE-2018-18310} {Low [] 2019:2197 CVE-2019-7665} {Moderate [] 2019:2237 CVE-2018-12404} {Moderate [] 2019:2237 CVE-2018-0495} {Moderate [] 2019:2304 CVE-2019-1559} {Moderate [] 2019:2304 CVE-2018-0734}] 52
```

**Comment by jasonmadigan** [comment]
@maleck13 looks good in any case and good to see the mechanism works. Tried it out and seems to work as expected.

**Review by jasonmadigan — APPROVED** [review]

**Pull request merged**